code/game/turfs/simulated/floor.dm

```diff
@@ -82,6 +82,22 @@ GLOBAL_LIST_INIT(icons_to_ignore_at_floor_init, list("damaged1","damaged2","dama
 		if(A.level == 3)
 			return 1
 
+// Checks if the turf is safe to be on
+/turf/simulated/floor/safety_turf()
+	if(islava(src))
```

Comment:
Instead of using a typecheck here you can better make an override for this proc on `/turf/simulated/floor/plating/lava`.
Be sure to call the parent as well to check the air content